### PR TITLE
Add table azure_container_group Closes #632

### LINKED ISSUE #632: Add table azure_container_group
This particular column is needed in a new [query](https://portal.azure.com/#view/Microsoft_Azure_Policy/PolicyDetailBlade/definitionId/%2Fproviders%2FMicrosoft.Authorization%2FpolicyDefinitions%2F0aa61e00-0a01-4a3c-9945-e93cffedf0e6) for Azure NIST SP 800 53 Rev 5 Updates : [Encryption Properties](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v58.0.0+incompatible/services/containerinstance/mgmt/2021-07-01/containerinstance#EncryptionProperties)

[List Call Link](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v58.0.0+incompatible/services/containerinstance/mgmt/2021-07-01/containerinstance#NewContainerGroupListResultPage)

[Terraform-Link](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/container_group)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  name,
  id,
  provisioning_state,
  restart_policy,
  sku,
  region
from
  azure_container_group;
+-------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+----------->
| name                    | id                                                                                                                                                    | provisioning_state | restart_po>
+-------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+----------->
| temp-container-instance | /subscriptions/gdaskask-hsjs-sha0-h989-809sax0sass/resourceGroups/demo/providers/Microsoft.ContainerInstance/containerGroups/temp-container-instance | Succeeded          | OnFailure >
| temp-test-pc            | /subscriptions/gdaskask-hsjs-sha0-h989-809sax0sass/resourceGroups/demo/providers/Microsoft.ContainerInstance/containerGroups/temp-test-pc            | Succeeded          | OnFailure >
+-------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+----------->
> select
  name,
  encryption_properties ->> 'VaultBaseURL' as vault_base_url,
  encryption_properties ->> 'KeyName' as key_name,
  encryption_properties ->> 'KeyVersion' as key_version,
  region
from
  azure_container_group;
+-------------------------+----------------+----------+-------------+--------+
| name                    | vault_base_url | key_name | key_version | region |
+-------------------------+----------------+----------+-------------+--------+
| temp-container-instance | <null>         | <null>   | <null>      | eastus |
| temp-test-pc            | <null>         | <null>   | <null>      | eastus |
+-------------------------+----------------+----------+-------------+--------+
> select
  name,
  restart_policy,
  provisioning_state,
  type
from
  azure_container_group
where
  restart_policy = "OnFailure";
Error: column "OnFailure" does not exist (SQLSTATE 42703)
> select
  os_type,
  count(name) as group_count
from
  azure_container_group
group by
  os_type;
+---------+-------------+
| os_type | group_count |
+---------+-------------+
| Linux   | 2           |
+---------+-------------+
> select
  name,
  ip_address -> 'Ports' as ports,
  ip_address ->> 'Type' as ip_address_type,
  ip_address ->> 'IP' as ip,
  ip_address ->> 'DNSNameLabel' as dns_name_label,
  ip_address ->> 'Fqdn' as fqdn
from
  azure_container_group;
+-------------------------+--------+-----------------+--------+----------------+--------+
| name                    | ports  | ip_address_type | ip     | dns_name_label | fqdn   |
+-------------------------+--------+-----------------+--------+----------------+--------+
| temp-container-instance | <null> | <null>          | <null> | <null>         | <null> |
| temp-test-pc            | <null> | <null>          | <null> | <null>         | <null> |
+-------------------------+--------+-----------------+--------+----------------+--------+
```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
Need to test the code changes.

File: azure/table_azure_container_group.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAzureContainerGroup(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "azure_container_group",
+		Description: "Azure Container Registry",
```
Comment (misraved) [inline azure/table_azure_container_group.go:17]:
```suggestion
		Description: "Azure Container Group",
```

File: azure/table_azure_container_group.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+				Transform:   transform.FromField("ContainerGroupProperties.OsType"),
+			},
+			{
+				Name:        "encryption_properties",
+				Description: "The encryption settings of container registry.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ContainerGroupProperties.EncryptionProperties"),
+			},
+			{
+				Name:        "Containers",
```
Comment (misraved) [inline azure/table_azure_container_group.go:77]:
```suggestion
				Name:        "containers",
```

File: azure/table_azure_container_group.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getContainerGroup(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	name := d.EqualsQuals["name"].GetStringValue()
+	resourceGroup := d.EqualsQuals["resource_group"].GetStringValue()
```
Comment (misraved) [inline azure/table_azure_container_group.go:225]:
```suggestion
	name := d. EqualsQualString("name")
	resourceGroup := d.EqualsQualString("resource_group")
```

File: docs/tables/azure_container_group.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  name,
+  encryption_properties ->> 'VaultBaseURL' as vault_base_url,
+  encryption_properties ->> 'KeyName' as key_name,
+  encryption_properties ->> 'KeyVersion' as key_version,
+  region
+from
+  azure_container_group;
+```
+
+### List groups that have retry policy set to OnFailure
```
Comment (misraved) [inline docs/tables/azure_container_group.md:34]:
```suggestion
### List groups that have restart policy set to `OnFailure`
```

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**